### PR TITLE
Dice mfg cmds

### PR DESCRIPTION
**Pull request by flihp**

### CODE REVIEW

**Comment by flihp** [comment]
AFAIK the most controversial part of this PR will be the new memory region where common artifacts are handed off. This requires that multiple tasks be able to read from the handoff memory region. Naturally write access will allow tasks to corrupt data consumed by another task. I've created an issue to track implementing a read-only mapping here: https://github.com/oxidecomputer/hubris/issues/835 and will take this work on if the design is acceptable.

File: lib/dice/src/mfg.rs
Comment (labbott) [inline lib/dice/src/mfg.rs:24]: Can you define ECA?

File: stage0/src/usart.rs
Comment (labbott) [inline stage0/src/usart.rs:48]: probably not worth turning off IOCON since the ROM ends up leaving it on for us anyway

File: stage0/src/usart.rs
Comment (labbott) [inline stage0/src/usart.rs:67]: again, probably not worth turning those off

File: lib/dice/src/mfg.rs
Comment (flihp) [reply lib/dice/src/mfg.rs:24]: Apologies, this is a DICE-ism. ECA is "embedded certificate authority" and in this case it's the DeviceId cert. These comments probably should have been removed before the PR & they'll be replaced by code in my next PR.

File: stage0/src/usart.rs
Comment (flihp) [reply stage0/src/usart.rs:67]: I figured the safest thing to do was to return the platform to the default state. This was intended to avoid possibly breaking assumptions made by existing hubris code but also to prevent future code in hubris relying on behavior in stage0 that may change. This may be overly cautious but unless we need to reduce code size or speed up boot time it seemed like the prudent thing to do.

Comment (flihp) [reply stage0/src/usart.rs:67]: LMK if you're asking for this to be removed before merging.

Comment (flihp) [reply stage0/src/usart.rs]: fixed

File: stage0/src/usart.rs
Comment (flihp) [reply stage0/src/usart.rs]: fixed

**Review by labbott — APPROVED** [review]
Overall LGTM

File: lib/dice/src/alias_cert_tmpl.rs
Comment (flihp) [reply lib/dice/src/alias_cert_tmpl.rs:5]: I doubt it. Looks like I got sloppy. Will fix this up before merging.

File: stage0/src/dice_mfg_usart.rs
Comment (flihp) [reply stage0/src/dice_mfg_usart.rs:32]: they will be missed

File: stage0/src/dice.rs
Comment (flihp) [reply stage0/src/dice.rs:21]: yup

File: lib/dice/src/alias_cert_tmpl.rs
Comment (flihp) [reply lib/dice/src/alias_cert_tmpl.rs:5]: ROFL: The code that generates this file doesn't generate the comment. So the template was updated but I didn't preserve the comment that was added by hand.

**Pull request merged**
